lib/linked_list.rb

```diff
@@ -7,34 +7,60 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def add_first(data)
-
+    if @head
+      @head = Node.new(data, @head)
+    else
+      @head = Node.new(data)
+    end
   end
-
-  # Time complexity - ?
-  # Space complexity - ?
+    
+  # Time complexity - O(1)
+  # Space complexity - O(1)
   def get_first
-
+    return @head.data if @head
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
-  def length
-    return 0
+  # Time complexity - O(n)
+  # Space complexity - O(1)
+  def length    
+    current = @head
+    count = 0
+
+    until current.nil?
+      count += 1
+      current = current.next
+    end
+
+    return count
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n)
+  # Space complexity - O(1)
   def add_last(data)
+    return add_first(data) if !@head
+  
+    last = @head.next
+    until last.nil?
+      last = last.next
+    end
 
+    Node.new(data, last)
```

Comment:
Right now you make a new node and make it's next node equal to last, which won't make it reachable. 

```suggestion
    return add_first(data) if !@head
  
    last = @head
    until last.next.nil?
      last = last.next
    end
    last.next = Node.new(data, nil)
```